angular/components/README.md

```diff
@@ -0,0 +1,153 @@
+# Components
+
+This page will give you some practices and tips/tricks about Angular components.
+
+## Clean Up Component to Prevent Memory Leak
+
+For example, when you create `interval` observable
+
+```ts
+import { interval, Subscription } from 'rxjs';
+
+@Component({...})
+export class MyComponent implements OnInit, OnDestroy {
+    ngOnInit () {
+        // create interval observable
+        const source = interval(60000);
+        source.subscribe(...);
+    }
+}
+```
+
+You should remember your observable and call `unsubscribe` function when component destroyed.
+
+```ts
+@Component({...})
+export class MyComponent implements OnInit, OnDestroy {
+    subscriptions: Subscription[] = []
+
+    ngOnInit () {
+        const source = interval(60000);
+
+        // remember subscription 
+        let subscription = source.subscribe(...);
+
+        // add subscription into array.
+        this.subscriptions.push(subscription);
+    }
+
+    // unsubscribed all subscriptions when component is destroyed.
+    ngOnDestroy() {
+        this.subscriptions.forEach((subscription) => subscription.unsubscribe());
+    }
+}
+```
+
+**Note** Angular will automatically clean up unsubscribe observable that created from `HttpClient`. More information [here](https://stackoverflow.com/questions/35042929/is-it-necessary-to-unsubscribe-from-observables-created-by-http-methods)
+
+## Do not Set HTML to Element Directly
+
+To prevent XSS Injections. Angular provides method to set HTML content.
+
+```ts
+// Bad
```

Comment:
I guess our conventions are `Yes`, `No`. What do you think krubbb?

Reply:
I agree krub.